flink-connectors/flink-connector-kinesis/src/test/java/org/apache/flink/streaming/connectors/kinesis/table/RowDataFieldsKinesisPartitionerTest.java

```diff
@@ -0,0 +1,241 @@
+
+	@Test
+	public void testBadPartitionerWithEmptyPrefix() {
+		thrown.expect(IllegalArgumentException.class);
+		thrown.expect(containsCause(new IllegalArgumentException(
+			"Cannot create a RowDataFieldsKinesisPartitioner for a non-partitioned table")));
+
+		CatalogTable table = createTable(defaultTableOptions(), Collections.emptyList());
+		new RowDataFieldsKinesisPartitioner(table);
+	}
+
+	@Test
+	public void testBadPartitionerWithDuplicateFieldNamesInPrefix() {
+		thrown.expect(IllegalArgumentException.class);
+		thrown.expect(containsCause(new IllegalArgumentException(
+			"The sequence of partition keys cannot contain duplicates")));
+
+		CatalogTable table = createTable(defaultTableOptions(), Arrays.asList("ip", "ip"));
+		new RowDataFieldsKinesisPartitioner(table);
+	}
+
+	@Test
+	public void testBadPartitionerWithBadFieldFieldNamesInPrefix() {
+		thrown.expect(IllegalArgumentException.class);
+		thrown.expect(containsCause(new IllegalArgumentException(
+			"The following partition keys are not present in the table: abc")));
+
+		CatalogTable table = createTable(defaultTableOptions(), Arrays.asList("ip", "abc"));
+		new RowDataFieldsKinesisPartitioner(table);
+	}
+
+	// --------------------------------------------------------------------------------------------
+	// Utilities
+	// --------------------------------------------------------------------------------------------
+
+	private RowData createElement(LocalDateTime time, String ip) {
+		return createElement(time, ip, "https://flink.apache.org/home");
+	}
+
+	private RowData createElement(LocalDateTime time, String ip, String route) {
+		GenericRowData element = new GenericRowData(TABLE_SCHEMA.getFieldCount());
+		element.setField(0, TimestampData.fromLocalDateTime(time));
+		element.setField(1, days(time));
+		element.setField(2, StringData.fromString(ip));
+		element.setField(3, StringData.fromString(route));
+		return element;
+	}
+
+	private int days(LocalDateTime time) {
+		return (int) ChronoUnit.DAYS.between(LocalDate.ofEpochDay(0), time);
+	}
+
+	private CatalogTable createTable(TableOptionsBuilder options, List<String> partitionKeys) {
+		return new CatalogTableImpl(TABLE_SCHEMA, partitionKeys, options.build(), TABLE_NAME);
+	}
+
+	private TableOptionsBuilder defaultTableOptions() {
+		String connector = KinesisDynamicTableFactory.IDENTIFIER;
+		String format = TestFormatFactory.IDENTIFIER;
+		return new TableOptionsBuilder(connector, format)
+			// default table options
+			.withTableOption(KinesisOptions.STREAM, TABLE_NAME)
+			.withTableOption("properties.aws.region", "us-west-2")
+			// default format options
+			.withFormatOption(TestFormatFactory.DELIMITER, ",");
+	}
+
+	/**
+	 * Repeat {@code n} times the string {@code s}.
+	 *
+	 * @param s The string to be repeated.
+	 * @param n The number of times to repeat the string.
+	 *
+	 * @return The repeated string.
+	 */
+	private static String repeat(String s, int n) {
```

Comment:
you can use `org.apache.flink.table.utils.EncodingUtils#repeat(java.lang.String, int)`

Reply:
Done.